rs/execution_environment/src/execution_environment/tests.rs

```diff
@@ -804,6 +804,177 @@ fn get_canister_status_of_nonexisting_canister() {
     assert_eq!(ErrorCode::CanisterNotFound, err.code());
 }
 
+fn get_canister_status(test: &mut ExecutionTest, canister_id: CanisterId) -> CanisterStatusResultV2 {
+    match test.canister_status(canister_id).unwrap() {
+        WasmResult::Reply(reply) => CanisterStatusResultV2::decode(&reply).unwrap(),
+        WasmResult::Reject(msg) => panic!("{}", msg) ,
+    }
+}
+
+#[test]
+fn get_canister_status_memory_metrics() {
+    let mut test = ExecutionTestBuilder::new().build();
+    let canister_id = test.universal_canister().unwrap();
+    let csr: CanisterStatusResultV2 = get_canister_status(&mut test, canister_id);
+    
+    let wasm_memory_size = csr.wasm_memory_size();
+    let stable_memory_size = csr.stable_memory_size();
+    let global_memory_size = csr.global_memory_size();
+    let wasm_binary_memory_size = csr.wasm_binary_memory_size();
+    let custom_sections_memory_size = csr.custom_sections_memory_size();
+    
+    let execution_memory_size = wasm_memory_size + stable_memory_size + global_memory_size + wasm_binary_memory_size + custom_sections_memory_size;
+
+    let canister_history_memory_size = csr.canister_history_memory_size();
+    let wasm_chunk_store_memory_size = csr.wasm_chunk_store_memory_size();
+    let snapshot_memory_size = csr.snapshot_memory_size();
+
+    let system_memory_size = canister_history_memory_size + wasm_chunk_store_memory_size + snapshot_memory_size;
+
+    let memory_size = csr.memory_size();
+    assert_eq!(memory_size, execution_memory_size + system_memory_size);
+}
+
+#[test]
+fn get_canister_status_memory_metrics_wasm_memory_size() {
+    let mut test = ExecutionTestBuilder::new().build();
+    let canister_id = test.universal_canister().unwrap();
+    let csr: CanisterStatusResultV2 = get_canister_status(&mut test, canister_id);
+
+    assert_eq!(
+        get_canister_status(&mut test, canister_id).wasm_memory_size(), 
+        csr.wasm_memory_size()
+    );
+
+    let canister_status_args = Encode!(&CanisterIdRecord::from(canister_id)).unwrap();
+    let get_canister_status = wasm()
+        .call_simple(
+            ic00::IC_00,
+            Method::CanisterStatus,
+            call_args().other_side(canister_status_args),
+        )
+        .build();
+    let result = test.ingress(canister_id, "update", get_canister_status);
+    let reply = get_reply(result);
+    let updated_csr = CanisterStatusResultV2::decode(&reply).unwrap();
+
+    assert_ne!(
+        updated_csr.wasm_memory_size(), 
+        csr.wasm_memory_size()
+    );
```

Comment:
Should we make this a bit more specific and check that it's greater than? We already use the [more_asserts](https://docs.rs/more-asserts/latest/more_asserts/macro.assert_gt.html) crate that provides some of these.